app/jobs/update_warehouse_clients_caches_job.rb

```diff
@@ -0,0 +1,39 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class UpdateWarehouseClientsCachesJob < BaseJob
+  queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+  WAIT_MINUTES = 4
+
+  def perform(client_ids: [], include_cas_and_cohorts: false, skip_expensive_calculations: false)
+    # If any for this class are already running, requeue for a few minutes in the future
+    any_jobs_of_type_other_than_self = Delayed::Job.jobs_for_class(self.class.name).
+      running.
+      where.not(Delayed::Job.arel_table[:handler].matches("%#{job_id}%")).
+      exists?
+    return requeue_job if any_jobs_of_type_other_than_self
+
+    GrdaWarehouse::WarehouseClientsProcessed.update_cached_counts(client_ids: client_ids, include_cas_and_cohorts: include_cas_and_cohorts, skip_expensive_calculations: skip_expensive_calculations)
+  end
+
+  private def requeue_job
```

Comment:
It would be nice to just use the built in `reschedule_at` but I'm betting it doesn't work us 
```ruby
def reschedule_at(time, _attempts)
  time + 4.minutes
end
```